pkg/cvo/sync_worker.go

```diff
@@ -535,128 +674,13 @@ func (w *SyncWorker) Status() *SyncWorkerStatus {
-			reporter.Report(SyncWorkerStatus{
-				Generation:  work.Generation,
-				Failure:     err,
-				Step:        "VerifyPayloadVersion",
-				Initial:     work.State.Initializing(),
-				Reconciling: work.State.Reconciling(),
-				Actual:      desired,
-				Verified:    info.Verified,
-			})
-			return err
-		}
-
-		// need to make sure the payload is only set when the preconditions have been successful
-		if len(w.preconditions) == 0 {
-			klog.V(4).Info("No preconditions configured.")
-		} else if info.Local {
-			klog.V(4).Info("Skipping preconditions for a local operator image payload.")
-		} else {
-			reporter.Report(SyncWorkerStatus{
-				Generation:  work.Generation,
-				Step:        "PreconditionChecks",
-				Initial:     work.State.Initializing(),
-				Reconciling: work.State.Reconciling(),
-				Actual:      desired,
-				Verified:    info.Verified,
-			})
-			if err := precondition.Summarize(w.preconditions.RunAll(ctx, precondition.ReleaseContext{DesiredVersion: payloadUpdate.Release.Version}, clusterVersion)); err != nil {
-				if work.Desired.Force {
-					klog.V(4).Infof("Forcing past precondition failures: %s", err)
-					w.eventRecorder.Eventf(cvoObjectRef, corev1.EventTypeWarning, "PreconditionsForced", "preconditions forced for payload loaded version=%q image=%q failures=%v", desired.Version, desired.Image, err)
-				} else {
-					w.eventRecorder.Eventf(cvoObjectRef, corev1.EventTypeWarning, "PreconditionsFailed", "preconditions failed for payload loaded version=%q image=%q failures=%v", desired.Version, desired.Image, err)
-					reporter.Report(SyncWorkerStatus{
-						Generation:  work.Generation,
-						Failure:     err,
-						Step:        "PreconditionChecks",
-						Initial:     work.State.Initializing(),
-						Reconciling: work.State.Reconciling(),
-						Actual:      desired,
-						Verified:    info.Verified,
-					})
-					return err
-				}
-			}
-			w.eventRecorder.Eventf(cvoObjectRef, corev1.EventTypeNormal, "PreconditionsPassed", "preconditions passed for payload loaded version=%q image=%q", desired.Version, desired.Image)
-		}
-
-		w.payload = payloadUpdate
-		w.eventRecorder.Eventf(cvoObjectRef, corev1.EventTypeNormal, "PayloadLoaded", "payload loaded version=%q image=%q", desired.Version, desired.Image)
-		klog.V(4).Infof("Payload loaded from %s with hash %s", desired.Image, payloadUpdate.ManifestHash)
-	}
-
-	return w.apply(ctx, w.payload, work, maxWorkers, reporter)
+// syncOnce applies the current payload to the server returning an error if
+// the update could not be completely applied. The status is updated as we
+// progress. Cancelling the context will abort the execution of syncOnce.
+func (w *SyncWorker) syncOnce(ctx context.Context, work *SyncWork, maxWorkers int, reporter StatusReporter) error {
+	klog.V(4).Infof("syncOnce: %s on generation %d in state %s at attempt %d", work.Desired.Version, work.Generation, work.State, work.Attempt)
```

Comment:
if this log line is the only remaining portion of `syncOnce` that's not in `apply`, maybe we should push it down into `apply` and drop `syncOnce`?